### PR TITLE
[V9] Port of account constraints

### PR DESCRIPTION
**Pull request by vrenaville**
It a WIP :
- I have remove check on the fiscalyear (useless now)
- I have remove cancel button on bank statement line, as in the futur we will be able to use account_draft_move or account_move_locking (installed with account_cancel) , for account_default_draft_move. 
- I have also add a check for payment order, since in V9 it used to generate move lines

### CODE REVIEW

**Comment by fclementic2c** [comment]
+1

File: account_constraints/__openerp__.py [hunk 1]
```diff
@@ -34,9 +34,6 @@
```
Comment (yvaucher) [inline account_constraints/__openerp__.py]: Description to move in a README.rst file

File: account_constraints/model/account_move_line.py [hunk 2]
```diff
@@ -45,12 +45,16 @@ def _authorized_reconcile(self, vals):
...
+            # Check if move related is linked to an invoice
+            invoice_linked = account_invoice_obj.search(
+                [('move_id', '=', line.move_id.id)])
+            if invoice_linked:
                 if self._authorized_reconcile(vals):
                     return True
                 err_msg = (_('Invoice name (id): %s (%s)') %
-                            (line.invoice.name, line.invoice.id))
+                            (invoice_linked.move_name, invoice_linked.id))
                 raise exceptions.Warning(
```
Comment (yvaucher) [inline account_constraints/model/account_move_line.py]: `Warning` is deprecated use `UserError` instead

File: account_constraints/model/account_move_line.py [hunk 3]
```diff
@@ -72,12 +76,28 @@ def _check_statement_related_move(self, vals=None):
...
 
+    @api.multi
+    def _check_payment_related_move(self, vals=None):
+        for line in self:
+            if line.payment_id:
+                if self._authorized_reconcile(vals):
+                    return True
+                err_msg = (_('Payment Order name (id): %s (%s)') %
+                            (line.payment_id.name, line.payment_id.id))
+                raise exceptions.Warning(
```
Comment (yvaucher) [inline account_constraints/model/account_move_line.py]: `Warning` is deprecated use `UserError` instead

**Comment by sbidoul** [comment]
Don't forget to remove view/account_bank_statement.xml.

And good luck with the test suite :smile_cat: 

Thanks for migrating this module!

File: account_constraints/__openerp__.py [hunk 4]
```diff
@@ -48,10 +45,9 @@
   if the same than the company one.
 
 * Remove the possibility to modify or delete a move line related to an
-  invoice or a bank statement, no matter what the status of the move
-  (draft, validated or posted). This is useful in a standard context but
-  even more if you're using `account_default_draft_move`. This way you ensure
-  that the user cannot make mistakes even in draft state, he must pass through
+  invoice or a bank statement or a payment order, no matter what the status of the move
```
Comment (elicoidal) [inline account_constraints/__openerp__.py]: remove 'what'

**Comment by yvaucher** [comment]
Readme.rst is not compliant with https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

Otherwise there is still some PEP8 to fix

**Pull request closed without merging**